doc/extending/cop.txt

```diff
@@ -186,6 +199,19 @@ There are less methods to define for an Op than for a Type:
        is high or when theano compilation directory is shared by many
        process (like on a network file server on a cluster).
 
+    .. method:: get_context(node)
+
+       (optional) If defined, should return the runtime context the op
+       needs.  This context will be passed to the C code through the
+       variable named in `sub['context']`.  The variable is also
+       available for use in the code returned by
+       :meth:`c_init_code_struct`.  If it returns `None` this is
+       considered the same as if the method was not defined.
```

Comment:
Is the doc up to date? If so, I still do not understand the implementation of run_context. run_context do an hasattr(.., 'get_context'), but to not make a special case if the call return None. Can you explain why this work? It would query a dict with None as the key and I do not see any thing that put None in the dict.